storage/Utils/Alignment.cc

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) [2016-2017] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact Novell, Inc.
+ *
+ * To contact Novell about this file by physical or electronic mail, you may
+ * find current contact information at www.novell.com.
+ */
+
+
+#include "storage/Utils/AlignmentImpl.h"
+
+
+namespace storage
+{
+
+    AlignError::AlignError()
+	: Exception("region cannot be aligned")
+    {
+    }
+
+
+    Alignment::Alignment(const Topology& topology)
+	: impl(new Impl(topology))
+    {
+    }
+
+
+    Alignment::Alignment(const Alignment& alignment)
+	: impl(new Impl(alignment.get_impl()))
+    {
+    }
+
+
+    Alignment::~Alignment()
```

Comment:
Kind of yes: The destructor of unique_ptr needs a complete type (see http://en.cppreference.com/w/cpp/memory/unique_ptr/%7Eunique_ptr), Alignment::Impl in this case. Since we do not provide AlignmentImpl.h in the API you would not be able to compile the class.